src/frontends/common/include/openvino/frontend/place.hpp

```diff
@@ -301,6 +312,25 @@ class FRONTEND_API Place {
     ///
     /// \param another Another place object
     virtual bool is_equal_data(const Ptr& another) const;
+
+    /// \brief Dynamic cast of place pointer.
+    ///
+    /// It should be used for Place sp casting instead of std::dynamic_pointer_cast as it check first
+    /// if pimpl is possible to cast.
+    ///
+    /// \tparam T     Result type of dynamic pointer cast.
+    ///
+    /// \param place  Input place sp.
+    ///
+    /// \return Casted sp if success otherwise nullptr.
+    template <class T>
+    static std::shared_ptr<T> dynamic_pointer_cast(const Ptr& place) {
```

Comment:
I always don't like to overload `dynamic_pointer_cast`....
Because user can use `dynamic_cast` and it won't work. Let's avoid it.